scripts/api.js

```diff
@@ -1,17 +1,30 @@
-/*global $ codeUrl */
-/* exported makeApiAjaxSettings */
+/*global codeUrl */
+/* exported ajax */
 
-function makeApiAjaxSettings(apiUrl, queryParams) {
-    let queryParamString = "";
-    if(queryParams) {
-        queryParamString = "&" + $.param(queryParams);
-    }
-    let settings = {
-        "url": codeUrl + "/api/index.php?url=" + apiUrl + queryParamString,
-        "dataType": "json",
-        "headers": {
-            "X-API-KEY": "SESSION"
-        }
+function ajax(method, apiUrl, queryParams = {}, data = {}) {
+    let url = new URL(`${codeUrl}/api/index.php`);
+    queryParams.url = apiUrl;
+    url.search = new URLSearchParams(queryParams);
+    let options = {
+        headers: {"X-API-KEY": "SESSION"},
+        method: method,
     };
-    return settings;
+    if(method !== "GET") {
```

Comment:
Might want to consider "get" lowercase too but not critical

Reply:
Rather than use the method as a parameter, is there a case for making separate functions for get, put and post? Then any mistakes would be caught by linting and wouldn't be left until run-time. It would also be better in that "get" can't have a body (or data parameter).

Reply:
Since this function is just for DP code -- and not for external systems using our code -- I don't think we need quite that level of abstraction. I do like doing a case-insensitive comparison for `get`/`GET`.

We should probably also be setting `options.headers['Accept'] = 'application/json';` for all request types as well.